lib/src/api/engine/remote/ws/native.rs

```diff
@@ -167,12 +167,7 @@ pub(crate) fn router(
 	route_rx: Receiver<Option<Route>>,
 ) {
 	tokio::spawn(async move {
-		let ping = {
-			let mut request = BTreeMap::new();
-			request.insert("method".to_owned(), PING_METHOD.into());
-			let value = Value::from(request);
-			Message::Binary(value.into())
-		};
+		let ping = Message::Ping(vec![]);
```

Comment:
@sgirones We are using the SurrealDB ping method here instead of the native WS ping because @tobiemh pointed out WS ping is not always available. This may not be relevant here because this is the native module, not the wasm one, but I'm not sure if there are any native platforms that don't support WS ping.

Reply:
oh interesting! thanks for pointing that out. @tobiemh how did you discover it? any steps to reproduce?

Reply:
the original reason for introducing this custom ping command was that certain Cloud providers will close the WebSocket connection unexpectedly (i.e. [see Google Cloud](https://cloud.google.com/load-balancing/docs/https#timeout-bes)), no matter the defined timeout.

Although I couldn't test it in practise, my assumption is that with this custom ping command the connection was reestablished automatically. Still, this is just masking the real problem, and is that if you want reliable long-lived WS connections, you need to put the proper layers in front.

As a counter example, AWS ALB support WebSockets too and only enforce connection-idle timeouts, so it should be better handled there.

I vote for going ahead with this change, as the WebSocket RFC only defines a pinger from server to client.

NOTE: after further discussion, intermediate proxies may as well cause issues. Still, we decided to go ahead and remove this higher level ping and rely on the built-in for now.

Reply:
More investigation took place, and it seems not having this higher level ping may be problematic when running the client in certain browsers. The connection may break for several reasons and the `on_close` event defined in the WebSockets API won't trigger until the connection is actually considered broken, most likely until a TCP timeout hits.

With a higher level ping, it forces the system to check the connection in order to send the message, and then it detects the connection is broken and triggers the necessary events.

Hence, we are adding the original ping back and will add a note in the code to remember in the future

Reply:
this should be resolved now